tests/bgp/conftest.py

```diff
@@ -128,3 +135,100 @@ def restore_nbr_gr(node=None, results=None):
 
     if not wait_until(300, 10, duthost.check_bgp_session_state, bgp_neighbors.keys()):
         pytest.fail("not all bgp sessions are up after disable graceful restart")
+
+
+@pytest.fixture(scope="module")
+def setup_interfaces(duthost, ptfhost, request, tbinfo):
+    """Setup interfaces for the new BGP peers on PTF."""
+
+    def _is_ipv4_address(ip_addr):
+        return ipaddress.ip_address(ip_addr).version == 4
+
+    @contextlib.contextmanager
+    def _setup_interfaces_t0(mg_facts, peer_count):
+        try:
+            connections = []
+            vlan_intf = None
+            for vlan_intf in mg_facts["minigraph_vlan_interfaces"]:
+                if _is_ipv4_address(vlan_intf["addr"]):
+                    break
+            if vlan_intf is None:
+                raise ValueError("No Vlan interface defined in T0.")
+            vlan_intf_name = vlan_intf["attachto"]
+            vlan_intf_addr = "%s/%s" % (vlan_intf["addr"], vlan_intf["prefixlen"])
+            vlan_members = mg_facts["minigraph_vlans"][vlan_intf_name]["members"]
+            local_interfaces = random.sample(vlan_members, peer_count)
+            neighbor_addresses = generate_ips(
+                peer_count,
+                vlan_intf["subnet"],
+                [netaddr.IPAddress(vlan_intf["addr"])]
+            )
+
+            for local_intf, neighbor_addr in zip(local_interfaces, neighbor_addresses):
+                conn = {}
+                conn["local_intf"] = vlan_intf_name
+                conn["local_addr"] = vlan_intf_addr
+                conn["neighbor_addr"] = neighbor_addr
+                conn["neighbor_intf"] = "eth%s" % mg_facts["minigraph_port_indices"][local_intf]
+                connections.append(conn)
+
+            for conn in connections:
+                ptfhost.shell("ifconfig %s %s" % (conn["neighbor_intf"],
+                                                  conn["neighbor_addr"]))
+
+            yield connections
+
+        finally:
+            for conn in connections:
+                ptfhost.shell("ifconfig %s 0.0.0.0" % conn["neighbor_intf"])
+
+    @contextlib.contextmanager
+    def _setup_interfaces_t1(mg_facts, peer_count):
+        try:
+            connections = []
+            ipv4_interfaces = [_ for _ in mg_facts["minigraph_interfaces"] if _is_ipv4_address(_['addr'])]
```

Comment:
The ```minigraph_interfaces``` contains only interfaces of DUT. The bgp neighbors also use some IP addresses. So will the generated IP addresses be conflict with the addresses of BGP neighbors?